src/Datadog.Trace.ClrProfiler.Managed/Emit/MethodBuilder.cs

```diff
@@ -351,15 +352,15 @@ private TDelegate EmitDelegate()
 
             if (methodInfo.ReturnType.IsValueType && returnType == typeof(object))
```

Comment:
nit: instead comparing `returnType == typeof(object)` we should compare if returnType is ValueType or not.

Reply:
I don't really want to change behavior right now, but this is some behavior that I think is fragile and requires us to specify return types correctly, which we have so far.